daemon/volumes.go

```diff
@@ -136,6 +136,9 @@ func (s *volumeStore) Create(name, driverName string, opts map[string]string) (v
 	if vc, exists := s.vols[name]; exists {
 		v := vc.Volume
 		s.mu.Unlock()
+		if driverName != v.DriverName() {
```

Comment:
This should go in daemon.VolumeCreate (in daemon/create.go) instead of here.
In cases where you have `docker run -v some_name:/foo --volume-driver other_driver` and `some_name` is an existing volume but on a different driver, this should work.